test/test_live_coinc_compare.py

```diff
@@ -0,0 +1,141 @@
+                "sigmasq": np.ones(self.num_trigs, dtype=np.float32),  # FIXME (maybe)
+                "template_id": np.random.uniform(
+                    0,
+                    self.num_templates,
+                    size=self.num_trigs
+                ).astype(np.int32)
+            }
+        self.start_time += self.analysis_chunk
+        return trigs
+
+
+class TestPyCBCLiveCoinc(unittest.TestCase):
+    def setUp(self, *args):
+        # Uncomment for more verbosity
+        # logging.basicConfig(format="%(asctime)s %(message)s",
+        #                     level=logging.INFO)
+
+        # simulate the `args` object we normally get from the command line arguments
+
+        url = 'https://github.com/gwastro/pycbc-config/raw/master/'
+        url += 'test_data_files/{}-PTA_HISTOGRAM.hdf'
+        stat_file_paths = [
+            download_file(url.format("H1L1"), cache=True),
+            download_file(url.format("H1V1"), cache=True),
+            download_file(url.format("L1V1"), cache=True)
+        ]
+        args = SimpleNamespace(
+            sngl_ranking="snr",
+            ranking_statistic="phasetd",
+            statistic_files=[stat_file_paths],
+            statistic_keywords=None,
+            timeslide_interval=0.1,
+            background_ifar_limit=100,
+            store_background=False
+        )
+
+        # number of templates in the bank
+        self.num_templates = 10
+
+        # duration of analysis segment
+        analysis_chunk = 2000
+
+        # combination of two detectors to analyze
+        detectors = ["H1", "L1"]
+
+        # number of single-detector triggers per detector per chunk
+        num_single_trigs = 400
+
+        self.num_iterations = 15
+
+        # create the single-detector trigger simulator
+        single_det_trig_sim = SingleDetTrigSimulator(
+            self.num_templates, analysis_chunk, detectors, num_single_trigs
+        )
+
+        self.new_trigs = [single_det_trig_sim.get_trigs()
+                          for _ in range(self.num_iterations)]
+
+        # create the current "coincer" object
+        self.new_coincer = Coincer.from_cli(args, self.num_templates,
+                                            analysis_chunk, detectors)
+
+        # create the validation "coincer" object
+        self.old_coincer = OriginalCoincer.from_cli(args, self.num_templates,
+                                                    analysis_chunk, detectors)
+
+
+    def test_coincer_runs(self):
+        # the following loop simulates the "infinite" analysis loop
+        # (though we only do a few iterations here)
+        for i in range(self.num_iterations):
+            logging.info("Iteration %d", i)
+            single_det_trigs = self.new_trigs[i]
+            coinc_trigs = self.new_coincer.add_singles(single_det_trigs)
+
+
+        for i in range(self.num_iterations):
+            logging.info("Iteration %d", i)
+            single_det_trigs = self.new_trigs[i]
```

Comment:
Why two loops?

Reply:
For debugging I found it easier to run the "old code" and then run the "new code", but I can edit this here.